code/modules/power/engines/tesla/energy_ball.dm

```diff
@@ -104,22 +108,35 @@
 	if(length(orbiting_balls))
 		. += "There are [length(orbiting_balls)] mini-balls orbiting it."
 
-/obj/singularity/energy_ball/proc/move_the_basket_ball(move_amount)
-	var/list/dirs = GLOB.alldirs.Copy()
-	if(length(shocked_things))
-		for(var/i in 1 to 30)
-			var/atom/real_thing = pick(shocked_things)
-			dirs += get_dir(src, real_thing) //Carry some momentum yeah? Just a bit tho
-	for(var/i in 0 to move_amount)
-		var/move_dir = pick(dirs) //ensures teslas don't just sit around
-		if(target && prob(10))
-			move_dir = get_dir(src,target)
-		var/turf/T = get_step(src, move_dir)
+/obj/singularity/energy_ball/proc/move_the_basket_ball(where_to_move)
+	if(locate(/obj/machinery/field/containment) in urange(15, src, 1)) // If there's a field in the closest 15 turfs we're sticking with a simple moving pattern
+		var/turf/T = get_step(src,pick(GLOB.alldirs))
 		if(can_move(T))
 			forceMove(T)
-			setDir(move_dir)
-			for(var/mob/living/carbon/C in loc)
-				dust_mobs(C)
+			return
+	else
+		if(!has_a_target)
+			find_the_basket()
+			return
+		else
+			for(var/i in 0 to 8)
+				forceMove(tesla_line[1])
+				tesla_line.Cut(1,2)
+				if(get_turf(src) == target_turf)
+					has_a_target = FALSE
+				for(var/mob/living/carbon/C in loc)
+					dust_mobs(C)
```

Comment:
Since you're already doing early returns technically, lets remove these redundant elses
```suggestion
	if(!has_a_target)
		find_the_basket()
		return
	for(var/i in 0 to 8)
		forceMove(tesla_line[1])
		tesla_line.Cut(1,2)
		if(get_turf(src) == target_turf)
			has_a_target = FALSE
		for(var/mob/living/carbon/C in loc)
			dust_mobs(C)
```